src/App.js

```diff
@@ -13,103 +16,38 @@ const StyledGlobal = createGlobalStyle`
 
 const StyledContainer = styled.div`
   display: flex;
-  flex-direction: column;
-  height: auto;
-  min-height: 100%;
-
-  background: #9bbbd4;
+  flex-direction: row;
 `;
 
-const userList = [
-  {
-    id: 0,
-    img: `${process.env.PUBLIC_URL}/muji.jpg`,
-    name: '무지',
-    status: '현재 접속 중',
-  },
-  {
-    id: 1,
-    img: `${process.env.PUBLIC_URL}/corn.png`,
-    name: '콘',
-    status: '현재 접속 중',
-  },
-];
-
-const sampleChat = [
-  {
-    user: userList[0],
-    chatList: ['안녕하세요'],
-  },
-  {
-    user: userList[1],
-    chatList: [
-      '안녕하세요',
-      '저도 반갑습니다',
-      `저는 ${userList[1].name}입니다.`,
-    ],
-  },
-  {
-    user: userList[0],
-    chatList: ['안녕하세요', '반갑습니다', `저는 ${userList[0].name}입니다.`],
-  },
-  {
-    user: userList[1],
-    chatList: [
-      '안녕하세요',
-      '저도 반갑습니다',
-      `저는 ${userList[1].name}입니다.`,
-    ],
-  },
-  {
-    user: userList[0],
-    chatList: [
-      '안녕하세요',
-      '반갑습니다',
-      `테스트 입력입니다테스트 입력입니다테스트 입력입니다테스트 입력입니다`,
-    ],
-  },
-  {
-    user: userList[1],
-    chatList: [
-      '안녕하세요',
-      '저도 반갑습니다',
-      `저는 ${userList[1].name}입니다.`,
-    ],
-  },
-];
-
 export default () => {
-  const [currentUser, setCurrentUser] = useState(userList[0]);
-  const [ownerUser, setOwnerUser] = useState(userList[0]);
-  const [chatData, setChatData] = useState(sampleChat);
-
-  const sendMessage = (message) => {
-    if (message === '') {
-      alert('빈 텍스트 입니다');
-      return;
-    }
-
-    const lastChatItem = chatData[chatData.length - 1];
-    if (lastChatItem.user.id === currentUser.id) {
-      lastChatItem.chatList.push(message);
-      setChatData([...chatData]);
-    } else {
-      const nextChatItem = { user: currentUser, chatList: [message] };
-      const newChatData = [...chatData, nextChatItem];
-      setChatData(newChatData);
-    }
-  };
-
-  const toggleCurrentUser = () => {
-    setCurrentUser(currentUser.id === 0 ? userList[1] : userList[0]);
-  };
+  const [chatList, setChatList] = useState(sampleChatList);
+  const [userList, setUserList] = useState(sampleUserList);
 
   return (
-    <StyledContainer>
+    <Router>
       <StyledGlobal />
-      <Header user={currentUser} onClickImage={toggleCurrentUser} />
-      <ChatContainer chatData={chatData} ownerUserId={ownerUser.id} />
-      <ChatInput onSendMessage={sendMessage} />
-    </StyledContainer>
+
+      <StyledContainer>
+        <Switch>
+          <Route
+            path={['/', '/user', '/chat', '/setting']}
+            component={Nav}
+            exact
+          />
+        </Switch>
+
+        <Switch>
+          <Route
+            path="/user"
+            render={(props) => <User {...props} userList={userList} />}
+          />
```

Comment:
```suggestion
          <Route
            path="/user"
          >
            <User userList={userList} />
          </Route>
```

혹시 이렇게 해도 작동 하던가요??
이렇게 하면 좀 더 직관적이고 좋을 것 같다는 생각이 들어용
prop으로 익명함수를 전달하는 것보다 children으로 넣어주는 패턴이 리액트의 설계의도와도 잘 맞을 듯 합니다 ㅎㅎ